repairnator/repairnator-scanner/src/main/java/fr/inria/spirals/repairnator/scanner/Launcher.java

```diff
@@ -346,6 +376,7 @@ private void printToStdout(Map<ScannedBuildStatus, List<BuildToBeInspected>> lis
 
     public static void main(String[] args) throws Exception {
         Launcher launcher = new Launcher(args);
+        ScannerBuildListener.getInstance().setLauncher(launcher);
```

Comment:
Usually we use the interface to declare an instance, the current line does not take advantages of polymorphism. For example:

```Java
BuildListener buildListener = null;
if (...) { // based on your arguments
    buildListener = new ActiveMQBuildListener();
} else if (...) {
    buildListener = new FileBuildListener();
}
if (buildListener != null) {
    buildListener.setLauncher(launcher);
    buildListener.runAsConsumerServer();
}
launcher.mainProcess();
```